src/utils.js

```diff
@@ -14,12 +17,20 @@ export function isWrappedModule(extensions) {
   }
 }
 
-export async function pathToRawSource(absPath) {
-  const fileURL = url.pathToFileURL(absPath);
+export async function pathToRawSource(p) {
+  const fileURL = url.pathToFileURL(p);
   return fs.promises.readFile(new URL(fileURL), 'utf8');
 }
 
-export function getSourceFromPathSync(absPath) {
-  const fileURL = url.pathToFileURL(absPath);
+export function getSourceFromPathSync(p) {
+  const fileURL = url.pathToFileURL(path.join(__root_dirname, p));
   return fs.readFileSync(new URL(fileURL), 'utf8');
 }
+
+export function isBareSpecifier(specifier) {
```

Comment:
Since this represents a standardized behavior (bare vs non-bare specifier test), it's often useful to link to the relevant spec so it's easier to verify / fix things in the future. In this case potentially this URL: https://html.spec.whatwg.org/multipage/webappapis.html#resolve-a-module-specifier

According to that spec, there's a few cases that would need to be tested that I don't think this implementation currently handles correctly. `.something` should be a bare specifier and `file:///tmp/x.mjs` shouldn't be one.

This could be fixed in a future PR that also adds a unit test for this, you did mention that the API for this isn't quite finalized.

Reply:
I've added a quick explanation for this above `isBareSpecifier`.  However, as you mentioned, this is not finalized and I will come back to this.  Thanks for the catch regarding those cases - I'll make sure there's unit tests for those once that API is finalized!